components/cluster/libqb/Makefile

```diff
@@ -40,16 +42,21 @@ CONFIGURE_OPTIONS += CFLAGS="$(CFLAGS)"
 CONFIGURE_OPTIONS += --enable-debug
 CONFIGURE_OPTIONS += --localstatedir=/var
 CONFIGURE_OPTIONS += --with-socket-dir=/var/run/hacluster
+CONFIGURE_OPTIONS += --enable-slow-tests
 
 COMPONENT_PREP_ACTION=(cd $(@D); $(ENV) $(CONFIGURE_ENV) $(CONFIG_SHELL) autogen.sh)
 
 COMPONENT_PRE_CONFIGURE_ACTION=($(CLONEY) $(SOURCE_DIR) $(@D)) 
 
-build:          $(BUILD_32)
+COMPONENT_PRE_TEST_ACTION += \
+    ($(MKDIR) /var/run/hacluster);
```

Comment:
this will not work, as non-root can't do this

Reply:
The value of --with-socket-dir defines SOCKETDIR in config.h and this variable is used in the source code.
The folder is created by svc-hacluster when the service will be enabled.
There is another PR https://github.com/grueni/oi-userland/blob/hacluster-common/components/cluster/hacluster-common/hacluster-common.p5m for hacluster-common.
The folder can be created in hacluster-common and libqb will be defined as dependent from hacluster-common.